### DISCUSSION TITLE
Add `eleventy --init` style command

### DISCUSSION THREAD

**Discussion by pdehaan** [11ty/eleventy#4081] [Enhancement Queue]
**Is your feature request related to a problem? Please describe.**
I create a lot of little 11ty projects and code repos, sometimes creating a basic scaffold can be tedious. Not sure if an `--init` flag would save me time by creating a basic .eleventy.js config file with a default input and output directory defined, and then pre-creating those directories would save me a lot of time and effort.

**Describe the solution you'd like**
<kbd>npx @11ty/eleventy --init</kbd> and it creates an .eleventy.js file which returns `{ dir: { input: "src", output; "_site" }};` and then creates an empty `./src/` directory (and possibly `./src/_includes/` and `./src/_data/`).

**Describe alternatives you've considered**
I can create my own little empty 11ty project and trigger it via npx, a la <kbd>npx pdehaan/create-11ty</kbd> (or <kbd>npm init pdehaan/11ty</kbd>) via https://github.com/pdehaan/create-11ty.

**Additional context**

**Comment by Ryuno-Ki** [comment]
Hm, that could lead to a Yeoman-kind of setup quickly?

**Comment by zachleat** [comment]
Just making sure you’ve seen the excellent https://github.com/philhawksworth/11up

**Comment by pdehaan** [comment]
I wasn't aware of 11up, but that's great news since I was planning on building something like that this weekend using [**degit**](https://www.npmjs.com/package/degit) (vs submodules).

**Comment by zachleat** [comment]
If we do end up adding something like this I think `npm init` is the way https://docs.npmjs.com/cli/v7/commands/npm-init

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by zachleat** [comment]
Some progress here in `@11ty/create` https://github.com/11ty/create

https://www.zachleat.com/web/smorgasbord-windows-terminal/
